controllers/system/system_controller.go

```diff
@@ -1724,12 +1724,26 @@ func (r *SystemReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	// The status reaches its desired status post reconciled
 	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&
```

Comment:
I truly hate the implementation of this block in: https://github.com/Wind-River/cloud-platform-deployment-manager/pull/352, unfortunately, I'm not confident to remove this whole block for now as DM migration concerns. Can you add a TODO above "if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&" on all controllers  to indicate we'll remove this block once no longer supporting the migration from helm application to FluxCD application? I'm Okay with this commit for now